src/container/container.tsx

```diff
@@ -19,6 +19,7 @@ import { TemplateInstance } from "./chart_template";
 
 export interface ChartContainerComponentProps {
   chart: Specification.Chart;
+  forceUpdate: (forceUpdate: () => void) => void;
```

Comment:
This feels a little awkward to me to expose a _basically_ internal function.

Which, I **_think_** is because when you update `this.chart`, you want the component re-render.

What about if you don't use this forceUpdate pass-through, but in `setChart`, instead of calling `this.forceUpdateCallback`, you call a new function called say, `render`

which would look like 

```
public render() {
   ReactDOM.render(this.reactMount(), this.container)
}
```

You also would update your `mount` function to call `this.render()` where you're currently doing the `ReactDOM.render`

Then, I think you can remove the `forceUpdate` code.

Which, admittedly is probably a little less efficient, but it doesn't expose internals.

What do you think?

Reply:
I agree. Thanks!

Reply:
I forget that In the Power BI visual the `this.container` is undefined.
Because Charticulator is typical react component in app and we have anchor element for whole app.

Reply:
Fixed the issue on visual side